src/java/org/apache/cassandra/config/Config.java

```diff
@@ -660,11 +661,13 @@
     public volatile boolean snapshot_on_repaired_data_mismatch = false;
 
     /**
-     * number of seconds to set nowInSec into the future when performing validation previews against repaired data
+     * Number of seconds to set nowInSec into the future when performing validation previews against repaired data
      * this (attempts) to prevent a race where validations on different machines are started on different sides of
      * a tombstone being compacted away
      */
-    public volatile int validation_preview_purge_head_start_in_sec = 60 * 60;
+
+    @Replaces(oldName = "validation_preview_purge_head_start_in_sec", converter = Converters.NEGATIVE_SECONDS_DURATION, deprecated = true)
+    public volatile SmallestDurationSeconds validation_preview_purge_head_start = new SmallestDurationSeconds("3600s");
```

Comment:
small bug in `Converters.NEGATIVE_SECONDS_DURATION`, should be `Integer.class` rather than `Long.class`

Reply:
In that sense we have also others that are Long.class in other converters and the primitive was int? Like `permissions_validity_in_ms` and its friends. Also, this un-converts  by using `DurationSpec::toSeconds` which returns long. Of course its getter in DD` getValidationPreviewPurgeHeadStartInSec` returns `conf.validation_preview_purge_head_start.toSecondsAsInt() ` or  
```
public int toSecondsAsInt()
    {
        return Ints.saturatedCast(toSeconds());
    }
```

Reply:
> In that sense we have also others that are Long.class in other converters and the primitive was int?

I should add a `diff` to detect this in CASSANDRA-17166.

> Also, this un-converts by using DurationSpec::toSeconds which returns long

Switching to `Integer` will make it no longer compile without returning `int`, so you would need to limit, such as calling `toSecondsAsInt`

Reply:
Added to `diff`, here is the set for 3.0

```
Replacement batch_size_fail_threshold is using a converter with old type java.lang.Long, but batch_size_fail_threshold_in_kb was int
Replacement batch_size_warn_threshold is using a converter with old type java.lang.Long, but batch_size_warn_threshold_in_kb was int
Replacement batchlog_replay_throttle is using a converter with old type java.lang.Long, but batchlog_replay_throttle_in_kb was int
Replacement cache_load_timeout is using a converter with old type java.lang.Long, but cache_load_timeout_seconds was int
Replacement column_index_size is using a converter with old type java.lang.Long, but column_index_size_in_kb was java.lang.Integer
Replacement commitlog_segment_size is using a converter with old type java.lang.Long, but commitlog_segment_size_in_mb was int
Replacement commitlog_sync_period is using a converter with old type java.lang.Long, but commitlog_sync_period_in_ms was java.lang.Integer
Replacement commitlog_total_space is using a converter with old type java.lang.Long, but commitlog_total_space_in_mb was java.lang.Integer
Replacement compaction_large_partition_warning_threshold is using a converter with old type java.lang.Long, but compaction_large_partition_warning_threshold_mb was java.lang.Integer
Replacement compaction_throughput is using a converter with old type java.lang.Long, but compaction_throughput_mb_per_sec was java.lang.Integer
Replacement dynamic_snitch_reset_interval is using a converter with old type java.lang.Long, but dynamic_snitch_reset_interval_in_ms was java.lang.Integer
Replacement dynamic_snitch_update_interval is using a converter with old type java.lang.Long, but dynamic_snitch_update_interval_in_ms was java.lang.Integer
Replacement file_cache_size is using a converter with old type java.lang.Long, but file_cache_size_in_mb was java.lang.Integer
Replacement gc_log_threshold is using a converter with old type java.lang.Long, but gc_log_threshold_in_ms was int
Replacement gc_warn_threshold is using a converter with old type java.lang.Long, but gc_warn_threshold_in_ms was int
Replacement hinted_handoff_throttle is using a converter with old type java.lang.Long, but hinted_handoff_throttle_in_kb was int
Replacement hints_flush_period is using a converter with old type java.lang.Long, but hints_flush_period_in_ms was int
Replacement index_summary_resize_interval is using a converter with old type java.lang.Long, but index_summary_resize_interval_in_minutes was int
Replacement inter_dc_stream_throughput_outbound is using a converter with old type java.lang.Long, but inter_dc_stream_throughput_outbound_megabits_per_sec was java.lang.Integer
Replacement internode_socket_receive_buffer_size is using a converter with old type java.lang.Long, but internode_recv_buff_size_in_bytes was java.lang.Integer
Replacement internode_socket_send_buffer_size is using a converter with old type java.lang.Long, but internode_send_buff_size_in_bytes was java.lang.Integer
Replacement max_hint_window is using a converter with old type java.lang.Long, but max_hint_window_in_ms was java.lang.Integer
Replacement max_hints_file_size is using a converter with old type java.lang.Long, but max_hints_file_size_in_mb was int
Replacement max_mutation_size is using a converter with old type java.lang.Long, but max_mutation_size_in_kb was java.lang.Integer
Replacement max_value_size is using a converter with old type java.lang.Long, but max_value_size_in_mb was java.lang.Integer
Replacement memtable_heap_space is using a converter with old type java.lang.Long, but memtable_heap_space_in_mb was java.lang.Integer
Replacement memtable_offheap_space is using a converter with old type java.lang.Long, but memtable_offheap_space_in_mb was java.lang.Integer
Replacement min_free_space_per_drive is using a converter with old type java.lang.Long, but min_free_space_per_drive_in_mb was java.lang.Integer
Replacement native_transport_max_frame_size is using a converter with old type java.lang.Long, but native_transport_max_frame_size_in_mb was java.lang.Integer
Replacement permissions_update_interval is using a converter with old type java.lang.Long, but permissions_update_interval_in_ms was int
Replacement permissions_validity is using a converter with old type java.lang.Long, but permissions_validity_in_ms was int
Replacement roles_update_interval is using a converter with old type java.lang.Long, but roles_update_interval_in_ms was int
Replacement roles_validity is using a converter with old type java.lang.Long, but roles_validity_in_ms was int
Replacement row_cache_size is using a converter with old type java.lang.Long, but row_cache_size_in_mb was long
Replacement sstable_preemptive_open_interval is using a converter with old type java.lang.Long, but sstable_preemptive_open_interval_in_mb was int
Replacement stream_throughput_outbound is using a converter with old type java.lang.Long, but stream_throughput_outbound_megabits_per_sec was java.lang.Integer
Replacement trace_type_query_ttl is using a converter with old type java.lang.Long, but tracetype_query_ttl was int
Replacement trace_type_repair_ttl is using a converter with old type java.lang.Long, but tracetype_repair_ttl was int
Replacement trickle_fsync_interval is using a converter with old type java.lang.Long, but trickle_fsync_interval_in_kb was int
```

Reply:
Yes, so more converters to handle the int/Integer

Reply:
So, I checked in diff and see that these converters are impacted


```
-    MILLIS_DURATION(Long.class,
+    MILLIS_DURATION(Integer.class, // needs int and long
-    SECONDS_DURATION(Long.class,
+    SECONDS_DURATION(Integer.class, // changed
-    MINUTES_DURATION(Long.class,
+    MINUTES_DURATION(Integer.class, // changed
-    MEBIBYTES_DATA_STORAGE(Long.class,
+    MEBIBYTES_DATA_STORAGE(Integer.class, // needs both int and long
-    KIBIBYTES_DATASTORAGE(Long.class,
+    KIBIBYTES_DATASTORAGE(Integer.class, // changed
-    BYTES_DATASTORAGE(Long.class,
+    BYTES_DATASTORAGE(Integer.class, // changed
-    MEBIBYTES_PER_SECOND_DATA_RATE(Long.class,
+    MEBIBYTES_PER_SECOND_DATA_RATE(Integer.class, // changed
-    MEGABITS_TO_MEBIBYTES_PER_SECOND_DATA_RATE(Long.class,
+    MEGABITS_TO_MEBIBYTES_PER_SECOND_DATA_RATE(Integer.class, // changed
-    MILLIS_CUSTOM_DURATION(Long.class,
+    MILLIS_CUSTOM_DURATION(Integer.class, // changed
```

So, fixing is low effort, but without the `diff` tests its painful..  I recommend we leave this alone in this patch and I fix it up in https://issues.apache.org/jira/browse/CASSANDRA-17166?

Reply:
As we agreed in Slack - leaving it to you in 17166, thanks!